### DISCUSSION TITLE
How to override the default Ryuk image?

### DISCUSSION THREAD

**Discussion by lalloni** [testcontainers/testcontainers-go#2044] [Q&A] [answered]
I tried both mechanisms for overriding image names (TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX env var and using an ImageSubstitutor implementation) to prefix the Ryuk image name with an internal registry URL. Both methods failed and were completely ignored for the Ryuk image. They worked for the other images we are running, though.

I think the issue is most likely caused by the fact that the default Ryuk image is defined as 

```go
ReaperDefaultImage = "docker.io/testcontainers/ryuk:0.5.1"
```

And that constant includes the `docker.io/` string prefix, making it non-overridable (according to documentation).

How should we do this?

If the right way of fixing the issue is changing the default image name to the following, I would be happy to send a PR:

```go
ReaperDefaultImage = "testcontainers/ryuk:0.5.1"
```

**Comment by mdelapenya** [comment]
Hey @lalloni I guess you are using `v0.26.0`, right? The hub prefix feature (https://github.com/testcontainers/testcontainers-go/pull/1928) is going to be released in the upcoming release.

Could you give it a try with the main branch? 🙏

**Reply by mdelapenya** [reply]
Mmmm wait, I know what you mean now: I've taken a look at the code and Ryuk code and it's exactly what you describe in the last sentence:

> that constant includes the docker.io/ string prefix, making it non-overridable (according to documentation).

I'm sending a PR ATM, thanks!

**Reply by mdelapenya** [reply]
https://github.com/testcontainers/testcontainers-go/issues/2045

**Reply by mdelapenya** [reply, marked as answer]
And PR here: https://github.com/testcontainers/testcontainers-go/pull/2046

Thank you for pointing this out!

I guess this is exactly your use case: ryuk running in your own registry, right?
